pkg/lineprinter/trimmer.go

```diff
@@ -0,0 +1,27 @@
+package lineprinter
```

Comment:
lineprinters is splitting on `\n` it should be the one trimming it. Why add something new?

Reply:
> lineprinters is splitting on `\n` it should be the one trimming it. Why add something new?

If you hook `LinePrinter` up to `fmt.Print`, you do want the trailing newlines:

```console
$ cat test.go
package main

import (
	"fmt"

	"github.com/openshift/installer/pkg/lineprinter"
)

func main() {
	linePrinter := &lineprinter.LinePrinter{
		Print: func (args ...interface{}) {
			fmt.Print(args...)
		},
	}
	linePrinter.Write([]byte("1\n2\n3\n4"))
	linePrinter.Close()
}
$ go run test.go 
1
2
3
4
```

Reply:
> > lineprinters is splitting on `\n` it should be the one trimming it. Why add something new?
> 
> If you hook `LinePrinter` up to `fmt.Print`, you do want the trailing newlines:
> 
> ```
> $ cat test.go
> package main
> 
> import (
> 	"fmt"
> 
> 	"github.com/openshift/installer/pkg/lineprinter"
> )
> 
> func main() {
> 	linePrinter := &lineprinter.LinePrinter{
> 		Print: func (args ...interface{}) {
> 			fmt.Print(args...)
> 		},
> 	}
> 	linePrinter.Write([]byte("1\n2\n3\n4"))
> 	linePrinter.Close()
> }
> $ go run test.go 
> 1
> 2
> 3
> 4
> ```

This example makes it seems like lineprinter is not named correctly then.

Reply:
> This example makes it seems like lineprinter is not named correctly then.

It reads in a stream and calls the wrapped print on each line.  What would you call it instead?